rust/src/client.rs

```diff
@@ -125,7 +129,9 @@ pub(crate) fn client_handle_fd_argument(arg: &str, arch: &str) -> CxxResult<Vec<
         return Ok(fds.into_iter().map(|f| f.into_raw_fd()).collect());
     }
 
-    if is_http_arg(arg) {
+    if is_src_rpm_arg(arg) {
+        return Err(anyhow!("src.rpm detected, please provide a valid rpm file for: {}", arg).into());
```

Comment:
One other nit on the phrasing here.  A `.src.rpm` *is* a valid rpm, it *can* be installed as root...but one almost never wants to do that.  (IMO we should entirely ditch `.src.rpm` in favor of git, but *that's* a whole other discussion :wink: )

How about something like:

```suggestion
        return Err(anyhow!("{} appears to be a source RPM which are not usually intended to be installed", arg).into());
```

Or maybe ```
        return Err(anyhow!("Refusing likely accidental requested installation of a source RPM: {}", arg).into());
```

?